commit fb631be6ce75
Author: DimiL <dlee@mozilla.com>
Date:   Wed Nov 1 08:40:00 2017 +0800

    Bug 1325888 - Clear cached completion result when close safebrowsing database. r=francois
    
    Safebrowsing testcase will close and then re-open database to simulate firefox restart.
    This bug happened when DB service already has a previously cached completion result and then
    re-open database. In this scenario, if testcase triggers another the same completion request,
    we won't cache it because we thought we have it already (See nsUrlClassifierDBServiceWorker::CacheCompletion).
    
    This causes a testcase assertion when testcase expects the completion
    result should be stored in LookupCache.
    
    MozReview-Commit-ID: 8o57jHv92OH
    
    --HG--
    extra : rebase_source : 90861060437d6cef860b85dd669d76e74ceec660
---
 toolkit/components/url-classifier/nsUrlClassifierDBService.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git toolkit/components/url-classifier/nsUrlClassifierDBService.cpp toolkit/components/url-classifier/nsUrlClassifierDBService.cpp
index e1a1be065aaf..a108812a1792 100644
--- toolkit/components/url-classifier/nsUrlClassifierDBService.cpp
+++ toolkit/components/url-classifier/nsUrlClassifierDBService.cpp
@@ -812,6 +812,12 @@ nsUrlClassifierDBServiceWorker::CloseDb()
     mClassifier = nullptr;
   }
 
+  // Clear last completion result when close db so we will still cache completion
+  // result next time we re-open it.
+  if (mLastResults) {
+    mLastResults->Clear();
+  }
+
   LOG(("urlclassifier db closed\n"));
 
   return NS_OK;
